### PR TITLE
[WIP] Updating TBB backend calls for tbb=2022.1

### PR DESCRIPTION
**Pull request by kozlov-alexey**
Adapts TBB parallel backend to API changes in tbb=2022.1 (task_scheduler_handle::get and task_scheduler_handle::release are removed)

### CODE REVIEW

File: numba/np/ufunc/tbbpool.cpp [hunk 1]
```diff
@@ -29,6 +30,12 @@ Implement parallel vectorize workqueue on top of Intel TBB.
  */
 #if TBB_INTERFACE_VERSION < 12010
 #error "TBB version is too old, 2021 update 1, i.e. TBB_INTERFACE_VERSION >= 12010 required"
+#elif TBB_INTERFACE_VERSION >= 12060
+#define TSH_ATTACH tbb::attach
```
Comment (Hardcode84) [inline numba/np/ufunc/tbbpool.cpp:34]: Can you make them functions instead of macros?

**Comment by Hardcode84** [comment]
Corresponding TBB PR https://github.com/oneapi-src/oneTBB/pull/566

File: numba/np/ufunc/tbbpool.cpp [hunk 2]
```diff
@@ -31,6 +32,24 @@ Implement parallel vectorize workqueue on top of Intel TBB.
 #error "TBB version is too old, 2021 update 1, i.e. TBB_INTERFACE_VERSION >= 12010 required"
 #endif
 
+tbb::task_scheduler_handle tbb_tsh_attach()
```
Comment (Hardcode84) [inline numba/np/ufunc/tbbpool.cpp:35]: static

**Comment by Hardcode84** [comment]
TBB version will be 2021.6

**Comment by sklam** [comment]
From OOB discussion, the new TBB *should* be ABI compatibility. Need to confirm on ABI compatibility.

**Comment by Hardcode84** [comment]
Some clarifications about ABI compatibility: forward compatibility (tbbpool built with new headers and + old `tbb.so`) is not guaranteed, backward compatibility (old headers + new `tbb.so`) is guaranteed and tested.

**Comment by stuartarchibald** [comment]
xref: https://github.com/conda-forge/numba-feedstock/pull/87#discussion_r784692198 when this patch is tested etc against 2021/2022, check back with CF to see if the repo metadata needs a fix based on any potential regressions/build changes etc.

**Comment by sklam** [comment]
TODO:

- [ ] test tbb-devel 2021.6 works with earlier tbb runtime

**Comment by stuartarchibald** [comment]
Closing this with view of using #8727. Thanks!

**Pull request closed without merging**